orttraining/orttraining/python/training/orttrainer.py

```diff
@@ -843,3 +853,368 @@ def _update_onnx_model_initializers(self, state_tensors):
+                    },
+                    "shared_optimizer_state":
+                    {
+                        type: dict,
+                        optional: True, # present optimizer is shared, absent otherwise.
+                        schema:
+                        {
+                            "step":
+                            {
+                                type: array,
+                            }
+                        }
+                    }
+                }
+            },
+            "trainer_options":
+            {
+                type: dict,
+                schema:
+                {
+                    "mixed_precision":
+                    {
+                        type: bool
+                    },
+                    "zero_stage":
+                    {
+                        type: int
+                    },
+                    "world_rank":
+                    {
+                        type: int
+                    },
+                    "world_size":
+                    {
+                        type: int
+                    },
+                    "optimizer_name":
+                    {
+                        type: str
+                    }
+                }
+            },
+            "partition_info":
+            {
+                type: dict,
+                optional: True, # present if states partitioned, else absent
+                schema:
+                {
+                    model_weight_name:
+                    {
+                        type: dict,
+                        schema:
+                        {
+                            "original_dim":
+                            {
+                                type: array
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        - When `pytorch_format = True`
+        schema:
+        {
+            model_weight_name:
+            {
+                type: tensor
+            }
+        }
+
+        Args:
+            pytorch_format: boolean flag to select either ONNX Runtime or PyTorch state schema
+
+        Returns:
+            A dictionary with `ORTTrainer` state
+        """
+        if not self._training_session:
+            warnings.warn("ONNX Runtime training session is not initialized yet. "
+                          "Please run train_step or eval_step at least once before calling ORTTrainer.state_dict().",
+                          UserWarning)
+            return self._load_state_dict.args[0] if self._load_state_dict else {}
+
+        state_dict = {}
+
+        # load training session model states into the state_dict
+        self._extract_model_states(state_dict, pytorch_format)
+        if pytorch_format:
+            # if pytorch_format is true, return a flat dictionary with only model states
+            # which is compatible with a PyTorch model
+            return state_dict[_utils.state_dict_model_key()]['fp32']
```

Comment:
what if the original model had a frozen fp16 weight, or one of any other data type? That will be excluded if we just return this way

Reply:
I will be putting all frozen weights in the ```fp32``` bucket. ```fp32``` would be implying the weights for the full precision model. Since the fp16 weights in this case would be frozen, they would be considered to be a part of the full precision model.